server/server.go

```diff
@@ -110,19 +110,19 @@ func (srv *Server) Flush() error {
 	return nil
 }
 
-func (srv *Server) Run() {
+func (srv *Server) Run(ctx context.Context) {
 	if srv.started {
 		log.Print("server already running")
 		return
 	}
 
 	srv.started = true
 	ticker := time.NewTicker(5 * time.Second)
-	done := make(chan bool)
 
 	for {
 		select {
-		case <-done:
+		case <-ctx.Done():
+			srv.Flush()
```

Comment:
I think you wanna close the underlying network connections. I forgot this is not http, during my latest review :)

E.g. you wanna call `srv.tcpParser.Close()` and the same for the ump connection.